wave_matching/src/icp.cpp

```diff
@@ -84,12 +130,152 @@ void ICPMatcher::estimateInfo() {
     switch (this->params.covar_estimator) {
         case ICPMatcherParams::covar_method::LUM: this->estimateLUM();
         case ICPMatcherParams::covar_method::CENSI: this->estimateCensi();
+        case ICPMatcherParams::covar_method::LUMold: this->estimateLUMold();
         default: return;
     }
 }
 
+void ICPMatcher::estimateLUMold() {
+    auto &source_trans = this->final;
+    PCLPointCloud *targetc;
+    if (this->params.res > 0) {
+        targetc = &(this->downsampled_target);
+    } else {
+        targetc = &(this->target);
+    }
+    int numSourcePts = source_trans->size();
+    std::vector<Eigen::Vector3f> corrs_aver(numSourcePts);
+    std::vector<Eigen::Vector3f> corrs_diff(numSourcePts);
+    int numCorr = 0;
+
+    wave::Mat6 edgeCov = wave::Mat6::Identity();
+
+    // build kd tree for source points
+    pcl::KdTreeFLANN<pcl::PointXYZ> kdtree;
+    if (this->params.res > 0) {
```

Comment:
This if statement doesn't do anything. Should it?

Reply:
no, it shouldn't exist